core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -240,31 +278,37 @@ object ZkUtils extends Logging {
   /**
    *  make sure a persistent path exists in ZK. Create the path if not exist.
    */
-  def makeSurePersistentPathExists(client: ZkClient, path: String) {
+  def makeSurePersistentPathExists(client: ZkClient, path: String, acls: List[ACL] = DefaultAcls) {
+    //Consumer path is kept open as different consumers will write under this node.
+    val acl = if(path == null || path.isEmpty || path.equals(ConsumersPath)) {
+      import scala.collection.JavaConversions._
+      ZooDefs.Ids.OPEN_ACL_UNSAFE.toList
+    } else acls
```

Comment:
/consumers should always be created as part of makeSurePersistentPathExists as this is created during server startup and I explain below why consumer path is kept open. I don't see why /consumer will be created as part of any of the other create*Path method which is why i did not pollute the code with getDefaultAclForPath() calls.